pubtools/_quay/operator_pusher.py

```diff
@@ -420,30 +429,64 @@ def build_index_images(self):
+            print([item.origin for item in items])
+            item_groups = {}
+            if is_hotfix and not is_advisory_source:
+                raise ValueError("Cannot push hotfixes without an advisory")
+            if is_hotfix:
+                for item in items:
+                    item_groups.setdefault(item.origin, []).append(item)
+            else:
+                item_groups["default"] = items
+
+            for group, g_items in item_groups.items():
+                if is_hotfix:
+                    tag = "{0}-{1}-{2}".format(
+                        version,
+                        items[0].metadata["com.redhat.hotfix"],
+                        items[0].origin.replace(":", "-"),
+                    )
+                    index_image = "{image_repo}:{tag}".format(
+                        image_repo=self.target_settings["iib_index_image"], tag=tag
+                    )
+                else:
+                    tag = version
+                    index_image = "{image_repo}:{tag}".format(
+                        image_repo=self.target_settings["iib_index_image"], tag=tag
+                    )
+
+                bundles = [self.public_bundle_ref(i) for i in g_items]
+                all_archs = [
+                    i.metadata["arch"] if i.metadata["arch"] != "x86_64" else "amd64"
+                    for i in g_items
+                ]
+                archs = sorted(list(set(all_archs)))
+                signing_keys = sorted(list(set([item.claims_signing_key for item in items])))
 
-            # Get deprecation list
-            deprecation_list = self.get_deprecation_list(version)
+                # Get deprecation list
+                deprecation_list = self.get_deprecation_list(version)
 
-            # build index image in IIB
-            index_image = "{image_repo}:{tag}".format(
-                image_repo=self.target_settings["iib_index_image"], tag=version
-            )
-            build_details = self.iib_add_bundles(
-                bundles=bundles,
-                archs=archs,
-                index_image=index_image,
-                deprecation_list=deprecation_list,
-                build_tags=["{0}-{1}".format(index_image.split(":")[1], self.task_id)],
-                target_settings=self.target_settings,
-            )
+                # build index image in IIB
+                if is_hotfix:
+                    override_settings = {"iib_overwrite_from_index": False}
+                else:
+                    override_settings = {}
+                build_details = self.iib_add_bundles(
+                    bundles=bundles,
+                    archs=archs,
+                    index_image=index_image,
+                    deprecation_list=deprecation_list,
+                    build_tags=["{0}-{1}".format(index_image.split(":")[1], self.task_id)],
+                    target_settings=self.target_settings,
+                    override_settings=override_settings,
```

Comment:
Wouldn't it be simpler to just copy target settings here and remove the desired parameters instead of introducing the strange new parameter "override_settings" to multiple methods?

Reply:
Original target settings shouldn't be changed. In that case you would need to copy it and do the changes on the copy. So I thought it's better to put there override

Reply:
Yeah, that's what I meant. Do you really think it's such a bad approach? I find override_settings very confusing. 

I mean, the function now has two parameters: a dictionary, and another dictionary to tell us how to modify the first dictionary. But the second parameter only works sometimes. I find this whole design very strange.